### DISCUSSION TITLE
background audio keeps playing even interstitial video ad is playing

### DISCUSSION THREAD

**Discussion by meehjeevan** [axmolengine/axmol#2011] [Q&A]
I'm playing a background music with this:
`AudioManager::getInstance()->playBackgroundMusic("bgmusic.mp3");`
but it keeps playing even an interstitial video ad is playing.. 
in my old version with cocosdension this issue wasn't occurring but now it's deprecated.

**Comment by GamesStudioTwo** [comment]
I'm sorry are you saying this is bug, I have not worked with adverts yet, I  thought you would have to detect them and pause what you are playing ?.

**Reply by meehjeevan** [reply]
I though this should have been managed with the audiomanager or the engine itself.. 
With cocos2dx 3.17 with cocosdension the bg music was stopping automatically but I found with AudioManager it's not working like that.

**Reply by GamesStudioTwo** [reply]
Ahh ok. it was a auto detect thing. there must be a way just to detect an advert. to then pause then play again ?

**Comment by rh101** [comment]
> in my old version with cocosdension this issue wasn't occurring but now it's deprecated.

You'll need to add your own management code for that.  The simple audio engine (cocosdension) was removed in Cocos2d-x v4, so it's not something that was removed in Axmol; it just wasn't there to begin with.
